Src/FluentAssertions/Events/EventHandlerFactory.cs

```diff
@@ -27,7 +27,7 @@ public static Delegate GenerateHandler(Type eventSignature, IEventRecorder recor
                 AppendParameterListThisReference(parameters),
                 module);
 
-            MethodInfo methodToCall = typeof(IEventRecorder).GetMethod("RecordEvent",
+            MethodInfo methodToCall = typeof(EventRecorder).GetMethod("RecordEvent",
```

Comment:
Does this silences the warning about `RecordEvent` being unused?
I assume that's why you added `[UsedImplicitly]`.

```suggestion
            MethodInfo methodToCall = typeof(EventRecorder).GetMethod(nameof(EventRecorder.RecordEvent),
```

Reply:
Yes.